### ISSUE TITLE
UI Error in Volume Creation: Incorrect Volume Name Limitation Warning on Hyphen Entry

### ISSUE THREAD

**Issue by wabernat** [scality/metalk8s#2339]
**Component**:

MetalK8s Dashboard

**What happened**:

While creating a volume, I named the volume using a hyphen, which is permissible. The UI raised a red warning: "Names must be lowercase letters, numbers, and hyphens."

**What was expected**:

Hyphens are legal. The warning should not appear. (The UI correctly accepts the hyphenated name when I click **Create**).

**Steps to reproduce**

1. Open Mk8s Platform
2. Click a node.
3. Click the **Volumes** tab.
4. Click the **+** button
5. Commit a data entry error that triggers "Please enter a name" (e.g., open the box and leave it for another).  
6. Enter "volume-" in the **Name** field.

**Resolution proposal** (optional):

Deactivate the warning on hyphen.

**Comment by ChengYanJin** [comment]
Hello William, 

Thanks for reporting the issue.

According to the [DNS label naming rule](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-label-names,) we should rename the volume `end with an alphanumeric character`. So we don't accept the name if it's the hyphens by the end.

However, maybe we should improve our error message to be more specific.

**Comment by wabernat** [comment]
Another possibility (I don't know if the logic is practical) would be to turn off this particular warning  unless/until someone actually tries to create a volume with a terminal hyphen (i.e., moves the live cursor outside the Volume Name field). In other words, I won't see this warning while actively entering a volume name either before (current behavior) or after making a mistake in any field.

**Comment by gdemonet** [comment]
Agreed, validation rules should only apply once the user submits the form, or at least starts filling in another field.
Not sure what the priority for this item is though (cc @thomasdanan).
